app/routes/__index/domains/new.tsx

```diff
@@ -1,11 +1,66 @@
-import { Heading } from '@chakra-ui/react';
+import { Container, Heading, Text } from '@chakra-ui/react';
+import { json, redirect } from '@remix-run/node';
+import { RecordType } from '@prisma/client';
+import { useActionData } from '@remix-run/react';
+import { z } from 'zod';
+import { parseFormSafe } from 'zodix';
+
+import DnsRecordForm from '~/components/dns-record/form';
+import { createRecord } from '~/models/record.server';
+import { requireUsername } from '~/session.server';
+
+import type { ActionArgs } from '@remix-run/node';
+import type { ZodError } from 'zod';
+
+function errorForField(error: ZodError, field: string) {
+  return error.issues.find((issue) => issue.path[0] === field)?.message;
+}
+
+export const action = async ({ request }: ActionArgs) => {
+  // Create a Zod schema for validation
+  // Optional is not needed as we get '' if nothing is entered
+  const DnsRecord = z.object({
+    name: z.string().min(1), // We do not want to consider '' a valid string
+    type: z.nativeEnum(RecordType),
+    value: z.string().min(1),
```

Comment:
In a follow-up, I bet we could do this with a function and some regex to check for different things based on the `type`.

Reply:
Created #277 for this and the [other comment](https://github.com/DevelopingSpace/starchart/pull/243#discussion_r1117072649) about using `<InputRightAddon>`